featuretools/entityset/deserialize.py

```diff
@@ -132,29 +144,56 @@ def read_entity_data(description, path):
 
 
 def read_data_description(path):
-    '''Read data description from disk.
+    '''Read data description from disk, S3 path, or URL.
 
         Args:
-            path (str): Location on disk to read `data_description.json`.
+            path (str): Location on disk, S3 path, or URL to read `data_description.json`.
 
         Returns:
             description (dict) : Description of :class:`.EntitySet`.
     '''
+
     path = os.path.abspath(path)
     assert os.path.exists(path), '"{}" does not exist'.format(path)
     file = os.path.join(path, 'data_description.json')
-    with open(file, 'r') as file:
+    with open(file, 'r', encoding='utf-8') as file:
         description = json.load(file)
     description['path'] = path
     return description
 
 
-def read_entityset(path, **kwargs):
-    '''Read entityset from disk.
+def read_entityset(path, profile_name=None, **kwargs):
+    '''Read entityset from disk, S3 path, or URL.
 
         Args:
-            path (str): Directory on disk to read `data_description.json`.
+            path (str): Directory on disk, S3 path, or URL to read `data_description.json`.
+            profile_name (str, bool): The AWS profile specified to write to S3. Will default to None and search for AWS credentials.
+                Set to False to use an anonymous profile.
             kwargs (keywords): Additional keyword arguments to pass as keyword arguments to the underlying deserialization method.
     '''
-    data_description = read_data_description(path)
-    return description_to_entityset(data_description, **kwargs)
+    if _is_url(path) or _is_s3(path):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            file_name = Path(path).name + ".tar"
+            file_path = os.path.join(tmpdir, file_name)
+            transport_params = {}
+            session = boto3.Session()
+            if isinstance(profile_name, str):
+                transport_params = {'session': boto3.Session(profile_name=profile_name)}
+            if session.get_credentials() is not None or profile_name is not False:
```

Comment:
If `profile_name` is `None` and `session.get_credentials()` is `None` the logic will try to use smartopen instead of s3fs.  Also if `profile_name` is `False` and `session.get_credentials()` finds credentials than it will use smartopen instead of s3fs.